code/modules/food_and_drinks/kitchen_machinery/deep_fryer.dm

```diff
@@ -43,6 +43,38 @@
 	var/obj/item/reagent_containers/food/snacks/deepfryholder/type = new(get_turf(src))
 	return type
 
+/obj/machinery/cooker/deepfryer/attackby(obj/item/I, mob/user, params)
+	if(istype(I, /obj/item/reagent_containers/glass) || istype(I, /obj/item/reagent_containers/food/drinks/ice))
+		if(I.reagents.has_reagent("ice"))
+			var/ice_amount = I.reagents.get_reagent_amount("ice")
+			I.reagents.remove_all(I.reagents.total_volume)
+			add_attack_logs(user, src, "poured [ice_amount]u ice into")
+			user.visible_message(
+				"<span class='warning'>[user] pours [I] into [src], and it seems to fizz a bit.</span>",
+				"<span class='warning'>You pour [I] into [src], and it seems to fizz a bit.</span>",
+				"You hear a splash, and a sizzle."
+			)
+
+			playsound(src, 'sound/goonstation/misc/drinkfizz.ogg', 25)
+			addtimer(CALLBACK(src, .proc/boil_step_1, user), 4 SECONDS)
+			addtimer(CALLBACK(src, .proc/boil_step_2, user, ice_amount), 5 SECONDS)
+
+			return TRUE
+
+	return ..()
+
+/obj/machinery/cooker/deepfryer/proc/boil_step_1(mob/user)
+	visible_message(
+		"<span class='danger'>[src] starts to bubble and froth unnervingly!</span>",
+		"<span class='danger'>You hear a growling and intimidating bubbling!</span>"
+	)
+
+	playsound(src, 'sound/machines/fryer/deep_fryer_emerge.ogg', 75)
+	to_chat(user, "<span class='userdanger'>Are you sure that was such a good idea?</span>")
+
+/obj/machinery/cooker/deepfryer/proc/boil_step_2(mob/user, ice_amount)
+	make_foam(ice_amount)
```

Comment:
Could you make the timer above call `make_foam` directly? Would remove the need for this proc.